debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/CollectionId.java

```diff
@@ -5,24 +5,28 @@
  */
 package io.debezium.connector.mongodb;
 
+import java.util.regex.Pattern;
+
 import io.debezium.annotation.Immutable;
 
 /**
  * A simple identifier for collections in a replica set.
- * 
+ *
  * @author Randall Hauch
  */
 @Immutable
 public final class CollectionId {
 
+    private static final Pattern IDENTIFIER_SEPARATOR_PATTERN = Pattern.compile("\\.");
+
     /**
      * Parse the supplied string, extracting the first 3 parts into a Collection.
-     * 
+     *
      * @param str the string representation of the collection identifier; may not be null
      * @return the collection ID, or null if it could not be parsed
      */
     public static CollectionId parse(String str) {
-        String[] parts = str.split("[\\" + '.' + "]");
+        String[] parts = IDENTIFIER_SEPARATOR_PATTERN.split(str);
```

Comment:
That's the only thing I'm wondering, what was the benefit of using the pattern `[\.]` over simply `\.`? I don't think using a character class makes a difference for a single char, but maybe there was some intention behind it, @rhauch?